packages/core/src/access.ts

```diff
@@ -121,14 +123,21 @@ export function canWriteResource(accessPolicy: AccessPolicy, resource: Resource)
 export function matchesAccessPolicy(accessPolicy: AccessPolicy, resource: Resource, readonlyMode: boolean): boolean {
   if (accessPolicy.resource) {
     for (const resourcePolicy of accessPolicy.resource) {
-      if (
-        matchesAccessPolicyResourcePolicy(
-          resource,
-          readonlyMode ? AccessPolicyInteraction.READ : AccessPolicyInteraction.UPDATE,
-          resourcePolicy
-        )
-      ) {
-        return true;
+      try {
+        if (
+          matchesAccessPolicyResourcePolicy(
+            resource,
+            readonlyMode ? AccessPolicyInteraction.READ : AccessPolicyInteraction.UPDATE,
+            resourcePolicy
+          )
+        ) {
+          return true;
+        }
+      } catch (err: unknown) {
+        // TODO(ThatOneBro 24 Oct 2024): Consider removing try-catch once invalid access policies are not allowed to be created (validating search parameters are valid)
+        console.error(
+          `[Access Policy]: Got error "${normalizeErrorString(err)}" while evaluating resource against ${getReferenceString(accessPolicy)}`
+        );
```

Comment:
I would suggest letting this `@medplum/core` function throw, and handle catching the error and logging specifically in the access policy logic in `@medplum/server`, but I'm content either way

Reply:
I think the problem is that users might be relying on calling this function directly in other places too and could suddenly start throwing on them 😓 

In principle I agree though. Thoughts @codyebberson, @mattlong , @rahul1  ?